sql/create.go

```diff
@@ -142,6 +142,8 @@ func (p *planner) CreateIndex(n *parser.CreateIndex) (planNode, error) {
 		return nil, err
 	}
 
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
right then. it seems I was very confused by the mechanics and thought the gateway was doing something it shouldn't have.
Anyway, I'll move this into the `client.Txn` object and store it inside the session differently.